flink-yarn/src/test/java/org/apache/flink/yarn/ContainerRequestReflectorTest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.yarn;
+
+import org.apache.flink.runtime.util.HadoopUtils;
+import org.apache.flink.util.TestLogger;
+
+import org.apache.hadoop.yarn.api.records.Priority;
+import org.apache.hadoop.yarn.api.records.Resource;
+import org.apache.hadoop.yarn.client.api.AMRMClient;
+import org.junit.Test;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+/** Tests for {@link ContainerRequestReflector}. */
+public class ContainerRequestReflectorTest extends TestLogger {
+
+    @Test
+    public void testGetContainerRequestIfConstructorPresent() {
+        final ContainerRequestReflector containerRequestReflector =
+                new ContainerRequestReflector(ContainerRequestWithConstructor.class);
+        Resource resource = Resource.newInstance(100, 1);
+        Priority priority = Priority.newInstance(1);
+
+        AMRMClient.ContainerRequest containerRequest =
+                containerRequestReflector.getContainerRequest(resource, priority, "GPU");
+        assertTrue(containerRequest instanceof ContainerRequestWithConstructor);
```

Comment:
I think we might not need to verify it. Same as below.

Reply:
I think the assertion is necessary that when if no node-label specified, the method of `getContainerRequest` in test case will return Hadoop original `ContainerRequest` object instead of `ContainerRequestWithConstructor` object.

Reply:
Ok. But `assertTrue(containerRequest instanceof AMRMClient.ContainerRequest);` is invalid and should be removed. Same as below.